src/openms/source/CHEMISTRY/AASequence.cpp

```diff
@@ -879,81 +879,100 @@ namespace OpenMS
     return os;
   }
 
-  String::ConstIterator AASequence::parseModRoundBrackets_(
-    const String::ConstIterator str_it, const String& str, AASequence& aas, bool dot_notation, bool dot_terminal)
+  String::ConstIterator AASequence::parseModRoundBrackets_(const String::ConstIterator str_it,
+                                                           const String& str,
+                                                           AASequence& aas,
+                                                           const ResidueModification::TermSpecificity& specificity)
   {
     OPENMS_PRECONDITION(*str_it == '(', "Modification must start with '('.");
+
     String::ConstIterator mod_start = str_it;
     String::ConstIterator mod_end = ++mod_start;
     Size open_brackets = 1;
     while (mod_end != str.end())
     {
-      if (*mod_end == ')') --open_brackets;
-      else if (*mod_end == '(') ++open_brackets;
-      if (!open_brackets) break;
+      if (*mod_end == ')')
+      {
+        --open_brackets;
+      }
+      else if (*mod_end == '(')
+      {
+        ++open_brackets;
+      }
+
+      if (!open_brackets)
+      {
+        break;
+      }
       ++mod_end;
     }
+
+    // Extract the actual modification as string
     std::string mod(mod_start, mod_end);
     if (mod_end == str.end())
     {
       throw Exception::ParseError(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, str,
           "Cannot convert string to peptide modification: missing ')'");
     }
+
     ModificationsDB* mod_db = ModificationsDB::getInstance();
-    if (aas.peptide_.empty()) // start of peptide -> N-terminal modification?
+
+    // start of peptide -> N-terminal modification?
+    if (aas.peptide_.empty() || specificity == ResidueModification::N_TERM) 
     {
-      aas.n_term_mod_ = &(mod_db->getModification(mod, "",
-                                                    ResidueModification::N_TERM));
+      aas.n_term_mod_ = &(mod_db->getModification(mod, "", ResidueModification::N_TERM));
       return mod_end;
     }
 
     const String& res = aas.peptide_.back()->getOneLetterCode();
+    if (specificity == ResidueModification::C_TERM) 
+    {
+      aas.c_term_mod_ = &(mod_db->getModification(mod, res, ResidueModification::C_TERM));
```

Comment:
@timosachsenberg I wonder whether we should really add the residue here, wouldnt it be more correct to do

```
aas.c_term_mod_ = &(mod_db->getModification(mod, "", ResidueModification::C_TERM));
```

or does it not matter since the residue is ignored in either case?

Reply:
Hmm I don't know but I recall that unimod has terminal modifications at any site and restricted to certain sites e.g.:
<umod:specificity hidden="0" site="C" position="Any N-term" classification="Artefact" spec_group="7">
and
<umod:specificity hidden="1" site="N-term" position="Any N-term"
                           classification="Chemical derivative"
                           spec_group="3"/>

maybe this code wants to assign a site specific modification (if possible)?

Reply:
I guess a use case might be e.g. a labeling combined with an enzyme that cuts at a AA and introduces the label.
Could be at residue or end group.